src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -38,10 +37,10 @@ public TypeFactory(OutputLibrary library, Type unknownType)
         /// </summary>
         /// <param name="inputType">The input type to convert.</param>
         /// <returns>The <see cref="CSharpType"/> of the input type.</returns>
-        public CSharpType CreateType(InputType inputType) => inputType switch
+        public CSharpType CreateType(InputType inputType, string? format = null) => inputType switch
```

Comment:
I think this is not a good practice.
Could we consolidate those xms format to corresponding input types?
If this is for DataFactory, we could create some new types such as `InputDataFactoryElementType` in the converter

Reply:
DataFacotry mdoels have been converted into InputGenericType in CodeModelConverter, `format` is needed for a few other cases

Reply:
Guess we can try to handle them in CodeModelConverter as well, will try it out when I resolve all the generation errors

Reply:
resolved